File: app/src/main/java/com/aquaguard/domain/usecase/GetLiveDeviceStatusUseCase.java
package com.aquaguard.domain.usecase;

import androidx.lifecycle.LiveData;

import com.aquaguard.domain.model.WaterReading;
import com.aquaguard.domain.repository.WaterRepository;

import javax.inject.Inject;

public class GetLiveDeviceStatusUseCase {
    private final WaterRepository waterRepository;

    @Inject
    public GetLiveDeviceStatusUseCase(WaterRepository waterRepository) {
        this.waterRepository = waterRepository;
    }

    public LiveData<WaterReading> execute(String deviceId) {
        return waterRepository.getLiveReading(deviceId);
    }
}

File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/domain/usecase/GetLiveDeviceStatusUseCase.kt
package com.aquaguard.domain.usecase

import com.aquaguard.domain.model.WaterReading
import com.aquaguard.domain.repository.WaterRepository
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class GetLiveDeviceStatusUseCase @Inject constructor(
    private val waterRepository: WaterRepository
) {
    operator fun invoke(deviceId: String): Flow<WaterReading?> {
        return waterRepository.getLiveReading(deviceId)
    }
}
